spec/lib/learn_test/js_strategy_spec.rb

```diff
@@ -0,0 +1,61 @@
+shared_examples_for "js_strategy" do
+  let(:runner) { double("Runner", options: {}) }
+  let(:strategy) { described_class.new(runner) }
+  let(:package) do
+    {
+      dependencies: {
+        dep1: "",
+        dep2: ""
+      },
+      devDependencies: {
+        devDep1: "",
+        devDep2: ""
+      },
+      peerDependencies: {
+        peerDep1: "",
+        peerDep2: ""
+      }
+    }
+  end
+
+  context "missing_dependencies?" do
+
+    before(:each) do
+      allow(File).to receive(:exist?).and_return(true)
+      allow(strategy).to receive(:js_package).and_return(package)
+    end
+
+    it "returns true if missing a dependency" do
+      allow(File).to receive(:exist?).with("node_modules/dep2").and_return(false)
+      expect(strategy.missing_dependencies?).to eq(true)
+    end
+
+    it "returns true if missing a devDependency" do
+      allow(File).to receive(:exist?).with("node_modules/devDep2").and_return(false)
+      expect(strategy.missing_dependencies?).to eq(true)
+    end
+
+    it "returns true if missing a peerDependency" do
+      allow(File).to receive(:exist?).with("node_modules/peerDep2").and_return(false)
+      expect(strategy.missing_dependencies?).to eq(true)
+    end
+
+    it "returns false if missing no dependencies" do
+      allow(File).to receive(:exist?).and_return(true)
+      expect(strategy.missing_dependencies?).to eq(false)
+    end
+
+    it "returns false if there are no dependencies" do
+      allow(strategy).to receive(:js_package).and_return({})
+      expect(strategy.missing_dependencies?).to eq(false)
+    end
+  end
+end
+
+describe LearnTest::Strategies::Mocha do
```

Comment:
I think we should avoid referencing `LearnTest::Strategies::Mocha` and `LearnTest::Strategies::Jest` in this test file. The `shared_examples_for`/`it_behaves_like` macros are cool, but I think I'd prefer to just create a generic class in this test file + `include` the `JsStrategy` module in it. Happy to hear other opinions though!

Reply:
Happy to talk about it - I'm not super familiar with `shared_examples_for` but this seems like the appropriate use case for it, wondering what cons you see using it? I can see adding specific tests to the mocha or jest strategies in the future in this same file, and since they both rely on js_strategies this seems like a pretty straightforward way to test the module itself without making up a class to do it.